### PR TITLE
Issue #664: Extracted and edited add-ons docs

### PR DESCRIPTION
**Pull request by thatdocslady**
Fixes issue #664 

This PR handles the add-ons docs:

1. Extracted the content from the "Managing Minishift" topic into a separate topic.
2. Light proofread and formatting fixes.

NOTE: I noticed that the terminology for applying/executing an add-on isn't consisten (for example, if you see PR #733 you will notice that the term used there is "execute" and in this topic we say "apply"). We should decide on a verb and stick with it. If it helps, "execute" implies that you start something and it keeps running until you stop it, and "apply" means a one-off action that leaves a permanent result that you don't need to touch again. Maybe we can even borrow the start/stop verbs from systemctl (especially since we already use enable/disable?

### CODE REVIEW

**Comment by hferentschik** [comment]
>  If it helps, "execute" implies that you start something and it keeps running until you stop it

I tend to disagree

**Comment by thatdocslady** [comment]
@hferentschik updated per feedback, you're right I forgot to update the topic map and now the index is also updated.

**Comment by hferentschik** [comment]
> updated per feedback, you're right I forgot to update the topic map and now the index is also updated.

and _layout.erb_ on _layouts_. If you don't want to do it, I can care of it.

File: docs/source/_topic_map.yml [hunk 1]
```diff
@@ -25,6 +25,8 @@ Topics:
         File: managing-minishift
       - Name: Interacting with Openshift
         File: interacting-with-openshift
+      - Name: Add-ons
```
Comment (hferentschik) [inline docs/source/_topic_map.yml]: This should go above "Interacting with OpenShift". In the order of things, it comes before interacting with OpenShift. That said, it should also be "OpenShift"

File: docs/source/using/addons.adoc [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+----
+$ minishift addons list --verbose=true
+----
+
+This command prints a list of installed add-ons. You should at least see the _anyuid_ addon listed.
+This is an important add-on that allows you to run images that do not use a pre-allocated UID.
+By default, this is not allowed in OpensShift.
+
+[[section-enabling-disabling-addons]]
+== Enabling and disabling addons
```
Comment (hferentschik) [inline docs/source/using/addons.adoc]: add-ons

File: docs/source/using/addons.adoc [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+Add-ons are enabled with the link:../command-ref/minishift_addons_enable{outfilesuffix}[`minishift addons enable`]
+command and disabled with the link:../command-ref/minishift_addons_disable{outfilesuffix}[`minishift addons disable`] command.
+The following examples show how to enable and disable the _anyuid_ add-on.
+
+[[example-enable-anyuid]]
+.Example: Enabling the anyuid addon
```
Comment (hferentschik) [inline docs/source/using/addons.adoc]: add-on

File: docs/source/using/addons.adoc [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+----
+
+This command extracts the default add-ons to the add-on installation directory
+(`$MINISHIFT_HOME/addons`). To view the list of installed add-ons, you can then run:
+
+----
+$ minishift addons list --verbose=true
+----
+
+This command prints a list of installed add-ons. You should at least see the _anyuid_ addon listed.
```
Comment (hferentschik) [inline docs/source/using/addons.adoc]: add-on

**Comment by thatdocslady** [comment]
@hferentschik updated.

File: docs/source/using/addons.adoc [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+useful in cases where you know that a command such as `oc` might take a few seconds before a certain
+resource can be queried.
+
+NOTE: Trying to add a command that is not listed will cause an error when the add-on gets parsed.
+
+[[addon-variable-interpolation]]
+=== Variable interpolation
+
+Minishift also allows the use of variables as part of the supported commands.
+Variables have the from _#{<variable-name>}_. <<example-addon-variable>>
```
Comment (hferentschik) [inline docs/source/using/addons.adoc]: This is another broken reference now

**Comment by hferentschik** [comment]
Merged, making an additional commit to fix the broken links.

**Pull request closed without merging**